### ISSUE TITLE
Error: `PushNotification.init` is not a function

### ISSUE THREAD

**Issue by greenlaw110** [driftyco/ng-cordova#1268]
I am using `push_v5` plugin and phonegap push plugin. I found it doesn't work on my ios app built by ionic ios platform. The error message is:

```
TypeError: PushNotification.init is not a function. (In 'PushNotification.init(t)', 'PushNotification.init' is undefined)
```

The relevant source code of `push_v5`:

![image](https://cloud.githubusercontent.com/assets/216930/15950097/3627d6c0-2eef-11e6-9120-306057becb3e.png)

I did a bit research on this issue and found there are two different phonegap push notification plugin:

1. https://github.com/phonegap/phonegap-plugin-push/
2. https://github.com/phonegap-build/PushPlugin

It looks like `push_v5` works with the first one, however we are using the second one which does not provide the `init` function instead this is what it did to initialize a `PushNotification` plugin:

![image](https://cloud.githubusercontent.com/assets/216930/15950178/da631768-2eef-11e6-97b6-e6f34ec835cd.png)

And the safari debug tool verified that:

![image](https://cloud.githubusercontent.com/assets/216930/15950192/f1d2567a-2eef-11e6-9644-799d2d03ed2b.png)

Question: does `push_v5` works with the `phonegap-build` version at all (if we make it pick up the `window.pushNotification` instance). Or are you going to support that?

**Comment by greenlaw110** [comment]
Okay, I am pretty sure `push_v5` doesn't work with the https://github.com/phonegap-build/PushPlugin. I got the new error after adding an `init` method to `PushNotification` module:

```
TypeError: r.on is not a function. (In 'r.on("registration",function(e){t.resolve(e.registrationId)})', 'r.on' is undefined)
```

**Comment by gortok** [comment]
You are correct; it works for the first plugin, not the second. If you want to create a patch that lets it work for both, I'll gladly accept a PR.

**Comment by frey1esm** [comment]
How was this solved? I am getting `uncaught TypeError: PushNotification.init is not a function`  while using the first plugin `https://github.com/phonegap/phonegap-plugin-push/`
